### PR TITLE
Add `user_state` endpoint for better detection on account suspension

### PR DESCRIPTION
**Pull request by MiddleRed**
**GET** `https://api.x.com/help-center/forms/api/prod/user_state.json`
Response:
```python
{"userState":"normal"}     # Normal account
# or
{"userState":"bounced"}    # Temporarily locked account
# or
{"userState":"suspended"}  # Suspened account
```
It should be helpful to detect if a failed read operation is caused by rate limit or account suspension limit.  
  
Also there is a commit including modification suggestions provided by Ruff , depends on you whether accept it or not.

## Summary by CodeRabbit

- **New Features**
  - Improved error handling with new exceptions for account status issues: `AccountLocked` and `AccountSuspended`.
  - Enhanced user state management with a new method to retrieve user state.

- **Bug Fixes**
  - Removed duplicate key in tweet data building function.
  - Simplified boolean checks in query building function.

- **Chores**
  - Cleaned up unused imports for better code maintainability.

### CODE REVIEW

File: twikit/__init__.py [hunk 1]
```diff
@@ -9,18 +9,42 @@
 
 __version__ = '1.7.1'
```
Comment (MiddleRed) [inline twikit/__init__.py:11]: https://docs.astral.sh/ruff/rules/unused-import/

File: twikit/utils.py [hunk 2]
```diff
@@ -675,13 +674,13 @@ def build_query(text: str, options: SearchOptions) -> str:
     if until := options.get('until'):
         text += f' until:{until}'
```
Comment (MiddleRed) [inline twikit/utils.py:676]: https://docs.astral.sh/ruff/rules/true-false-comparison

**Comment by d60** [comment]
@MiddleRed 
Thank you for the pull request!
is there a specific reason for writing “import X as X” in the __init__.py file?

**Comment by MiddleRed** [comment]
> @MiddleRed Thank you for the pull request! is there a specific reason for writing “import X as X” in __init__.py?

https://docs.astral.sh/ruff/rules/unused-import/
I'm just applying Ruff's suggestions, it is finally up to you to decide to accept it or not

**Comment by d60** [comment]
@MiddleRed 
Understood. I prefer the previous style, so please revert the code.

**Comment by MiddleRed** [comment]
here you go

**Pull request merged**